homeassistant/components/backup/config.py

```diff
@@ -2,56 +2,116 @@
 
 from __future__ import annotations
 
-from dataclasses import dataclass
+from dataclasses import dataclass, field, replace
 from datetime import datetime
-from typing import Self, TypedDict
+from typing import TYPE_CHECKING, Self, TypedDict
 
-from homeassistant.core import HomeAssistant
+from cronsim import CronSim
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.event import async_track_point_in_time
 from homeassistant.helpers.storage import Store
 from homeassistant.helpers.typing import UNDEFINED, UndefinedType
+from homeassistant.util import dt as dt_util
 
 from .const import DOMAIN
 
+if TYPE_CHECKING:
+    from .manager import BackupManager
+
+CRON_PATTERN_DAILY = "0 2 * * *"
+CRON_PATTERN_WEEKLY = "0 2 * * {}"
```

Comment:
We should add a comment saying this should be aligned with, and happen after, recorder's nightly job.
I don't think we need to wait for the recorder job to finish, but we should schedule it 30 minutes later for example.

Reply:
Recorder runs its job at 04:12: <https://github.com/home-assistant/core/blob/f3733756494e307d9d951346ddf396a1b66edb8e/homeassistant/components/recorder/core.py#L646-L649>, the 12 minute offset is in case users have other hourly jobs scheduled.

Maybe we should schedule our job at 04:45? That would also make us safe against fold issues when there is no 02:00 or it's 02:00 twice at DST change.